packages/bundler-plugin-core/src/sentry/releasePipeline.ts

```diff
@@ -99,22 +59,16 @@ export async function setCommits(options: InternalOptions, ctx: BuildContext): P
   if (options.setCommits) {
     const { auto, repo, commit, previousCommit, ignoreMissing, ignoreEmpty } = options.setCommits;
 
-    if (auto || (repo && commit)) {
-      await ctx.cli.releases.setCommits(options.release, {
```

Comment:
After seeing that the unit tests fail, I'm realizing that although we can rely on earlier validation, this check here still served a purpose for this unit of code... (hence the unit tests that check for the portions of invalid option here and in `addDeploy` fail). 

Which makes me think... do we prefer central validation (like in this PR currently) or local validation? I could also refactor this and only validate in the respective steps. Open for opinions on this before I adapt tests.

Reply:
I like the idea of global validation, because then we can also adjust our types as it flows through the cli pipeline. For example, `validateOptions` could return a type predicate that states what kind of exact fields are defined on the options object.

Reply:
I don't have a strong preference either way. IMHO central validation is fine here and maybe a bit easier to follow, but I'm ok either way.

Reply:
After discussing this offline, we settled on global validation but with limited TS checks because in this particular situation it's quite tricky and would probably require complex typing. 

I made however a small adjustment to the user-facing `setCommits` type to make it clearer that `auto` and (`repo` && `commit`) are supposed to be mutually exclusive. Nevertheless, I left the checks in place and added a warning in case all three options are provided.

Will mark this PR as ready to review once #104 is merged and I rebased